bundles/core/src/main/java/com/adobe/cq/email/core/components/models/ContainerModel.java

```diff
@@ -0,0 +1,101 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2021 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.email.core.components.models;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Default;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.Optional;
+
+@Model(adaptables = Resource.class)
+public class ContainerModel {
+
+    @Inject
+    @Optional
+    @Default(values = "6")
+    private String layout;
+
+    public String getColClass1() {
+        return colClass1;
+    }
+
+    public String getColClass2() {
+        return colClass2;
+    }
+
+    public String getColClass3() {
+        return colClass3;
+    }
+
+    public int getColumns() {
+        return columns;
+    }
+
+    public String getLayout() {
+        return layout;
+    }
+
+    private String colClass1;
+    private String colClass2;
+    private String colClass3;
+    private int columns;
+
+
+    @PostConstruct
+    private void initModel() {
+        columns = 1;
+        initializeGrid();
+        buildClass();
+    }
+
+    private void initializeGrid() {
+        switch (layout) {
+            case "6":
+                columns = 1;
+                break;
+            case "3-3":
```

Comment:
The three cases 3-3, 2-4 and 4-2 should share their code block and do not leave the default empty, it should show the same behaviour as the default value for columns (6).

Reply:
Fixed